lib/daru_lite/dataframe.rb

```diff
@@ -1157,6 +1157,35 @@ def order=(order_array)
       initialize(to_h, order: order_array)
     end
 
+    # Returns the dataframe with reordered rows according to the index order given
+    # @param new_order => [Array], the new order you want for your rows
+    # @example
+    #   df = DaruLite::DataFrame.new({
+    #         a: [1, 2, 3],
+    #         b: [4, 5, 6]
+    #         },
+    #         order: [:a, :b]
+    #       )
+    #
+    #   df = #<DaruLite::DataFrame(3x2)>
+    #   #       a   b
+    #   #   0   1   4
+    #   #   1   2   5
+    #   #   2   3   6
+    #
+    #   df.reorder_rows[2,0,1]
+    #   df
+    #   # =>  #<DaruLite::DataFrame(3x2)>
+    #   #       a   b
+    #   #   2   3   6
+    #   #   0   1   4
+    #   #   1   2   5
+    def reorder_rows(new_order = [])
+      raise NotImplementedError if index.is_a?(DaruLite::CategoricalIndex)
+
+      row_at(*new_order)
```

Comment:
why create a new method, if #row_at already handle the job ?

Reply:
#row_at seems to works for what we want to do here, should I rename the method to make it more understandable (since we did not understood clearly what it does) ?